src/Sylius/Abstraction/StateMachine/src/WinzouStateMachineAdapter.php

```diff
@@ -42,21 +42,85 @@ public function apply(object $subject, string $graphName, string $transition, ar
     }
 
     public function getEnabledTransitions(object $subject, string $graphName): array
+    {
+        $stateMachine = $this->getStateMachine($subject, $graphName);
+
+        return array_filter(
+            $this->getAllTransitions($stateMachine),
+            fn (TransitionInterface $transition) => $this->can($subject, $graphName, $transition->getName()),
+        );
+    }
+
+    /**
+     * @return array<TransitionInterface>
+     */
+    private function getAllTransitions(\SM\StateMachine\StateMachineInterface $stateMachine): array
     {
         try {
-            $transitions = $this->getStateMachine($subject, $graphName)->getPossibleTransitions();
-        } catch (SMException $exception) {
+            $transitionsConfig = $this->getConfig($stateMachine)['transitions'];
+        } catch (\ReflectionException $exception) {
             throw new StateMachineExecutionException($exception->getMessage(), $exception->getCode(), $exception);
         }
 
-        return array_map(
-            fn (string $transition) => new Transition($transition, null, null),
-            $transitions,
-        );
+        $transitions = [];
+
+        foreach ($transitionsConfig as $transitionName => $transitionConfig) {
+            $froms = $transitionConfig['from'];
+            $tos = array($transitionConfig['to']);
+            $transitions[] = new Transition($transitionName, $froms, $tos);
+        }
+
+        return $transitions;
+    }
+
+    /**
+     * @throws \ReflectionException
+     *
+     * @return array{transitions: array<string, array{from: array<string>, to: string}>}
+     */
+    private function getConfig(\SM\StateMachine\StateMachineInterface $stateMachine): array
+    {
+        $reflection = new \ReflectionClass($stateMachine);
+        $configProperty = $reflection->getProperty('config');
+        $configProperty->setAccessible(true);
+
+        return  $configProperty->getValue($stateMachine);
+    }
+
+    /**
+     * @throws StateMachineExecutionException
+     */
+    public function getTransitionFromState(object $subject, string $graphName, string $fromState): ?string
```

Comment:
public before private :smile:

Reply:
It's discussable. I've talked with Janek about it, and we agreed we should all talk about it. I'm in favor of leaving it as-is, and eventually change it later.

Reply:
We'll discuss it further at a later date, either way, if we'd want to revert it and keep the public => protected => private order that'll be fixed along with adding an ecs rule.